c/authService.c

```diff
@@ -173,6 +174,86 @@ static int serveAuthCheck(HttpService *service, HttpResponse *res) {
   return 0;
 }
 
+int resetPassword(HttpService *service, HttpResponse *response) {
+  int returnCode = 0, reasonCode = 0;
+  HttpRequest *request = response->request;
+  
+  if (!strcmp(request->method, methodPOST)) {
+    char *inPtr = request->contentBody;
+    char *nativeBody = copyStringToNative(request->slh, inPtr, strlen(inPtr));
+    int inLen = nativeBody == NULL ? 0 : strlen(nativeBody);
+    char errBuf[JSON_ERROR_BUFFER_SIZE];
+    char responseString[RESPONSE_MESSAGE_LENGTH];
+
+    Json *body = jsonParseUnterminatedString(request->slh, nativeBody, inLen, errBuf, JSON_ERROR_BUFFER_SIZE);
+    JsonObject *inputMessage = jsonAsObject(body);
+    Json *username = jsonObjectGetPropertyValue(inputMessage,"username");
+    Json *password = jsonObjectGetPropertyValue(inputMessage,"password");
+    Json *newPassword = jsonObjectGetPropertyValue(inputMessage,"newPassword");
+    int usernameLength = 0;
+    int passwordLength = 0;
+    int newPasswordLength = 0;
+    if(username != NULL) {
+      usernameLength = strlen(jsonAsString(username));
+    }
+    if(password != NULL) {
+      passwordLength = strlen(jsonAsString(password));
+    }
+    if(newPassword != NULL) {
+      newPasswordLength = strlen(jsonAsString(newPassword));
+    }
+    if (usernameLength == 0) {
+      respondWithJsonStatus(response, "No username provided",
+                            HTTP_STATUS_BAD_REQUEST, "Bad Request");
+      return HTTP_SERVICE_FAILED;
+    } else if (passwordLength == 0) {
+      respondWithJsonStatus(response, "No password provided",
+                            HTTP_STATUS_BAD_REQUEST, "Bad Request");
+      return HTTP_SERVICE_FAILED;
+    } else if (newPasswordLength == 0) {
+      respondWithJsonStatus(response, "No new password provided",
+                            HTTP_STATUS_BAD_REQUEST, "Bad Request");
+      return HTTP_SERVICE_FAILED;
+    } else {
```

Comment:
This is not a bug, just a note on the structure. Since every check results in leaving the function, you can make all these `if-else`s `if`s, and remove the last `else` which will make the code much more readable.